services/app-api/src/postgres/contractAndRates/parseContractWithHistory.ts

```diff
@@ -404,14 +371,60 @@ function contractWithHistoryToDomainModel(
         stateCode: contract.stateCode,
         stateNumber: contract.stateNumber,
         draftRevision,
-        draftRates,
         revisions: revisions.reverse(),
         packageSubmissions: packageSubmissions.reverse(),
     }
 }
 
+// contractWithHistoryToDomainModel constructs a history for this particular contract including changes to all of its
+// revisions and all related rate revisions, including added and removed rates
+function contractWithHistoryToDomainModel(
+    contract: ContractTableFullPayload
+): ContractType | Error {
+    const contractWithoutRates =
+        contractWithHistoryToDomainModelWithoutRates(contract)
+    if (contractWithoutRates instanceof Error) {
+        return contractWithoutRates
+    }
+
+    if (
+        contractWithoutRates.status === 'SUBMITTED' ||
+        contractWithoutRates.status === 'RESUBMITTED'
+    ) {
+        return contractWithoutRates
+    }
+
+    // since we have a draft revision, we should also hold onto any set draftRates for later
+    const draftRatesOrErrors = contract.draftRates.map((dr) =>
+        rateWithoutDraftContractsToDomainModel(dr.rate)
+    )
+
+    const draftRatesOrError = arrayOrFirstError(draftRatesOrErrors)
+    if (draftRatesOrError instanceof Error) {
+        return draftRatesOrError
+    }
+
+    // FIX PARENT ID IF NEEDED
+    // This is fragile code. Because of the asymmetry required to break the
```

Comment:
seems like then we should write a unit test at least for this - make sure that placeholder is never showing up